exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java

```diff
@@ -239,6 +250,119 @@ private static Attributes toAttributesProto(
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "instance_id",
+            awsEc2InstanceMonitoredResource.getInstanceId());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "region",
+            "aws:" + awsEc2InstanceMonitoredResource.getRegion());
+        return;
+      case GCP_GCE_INSTANCE:
+        @SuppressWarnings("unchecked")
+        GcpGceInstanceMonitoredResource gcpGceInstanceMonitoredResource =
+            (GcpGceInstanceMonitoredResource) resource;
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "project_id",
+            gcpGceInstanceMonitoredResource.getAccount());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "instance_id",
+            gcpGceInstanceMonitoredResource.getInstanceId());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder, resourceType, "zone", gcpGceInstanceMonitoredResource.getZone());
+        return;
+      case GCP_GKE_CONTAINER:
+        @SuppressWarnings("unchecked")
+        GcpGkeContainerMonitoredResource gcpGkeContainerMonitoredResource =
+            (GcpGkeContainerMonitoredResource) resource;
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "project_id",
+            gcpGkeContainerMonitoredResource.getAccount());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "instance_id",
+            gcpGkeContainerMonitoredResource.getInstanceId());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder, resourceType, "zone", gcpGkeContainerMonitoredResource.getZone());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "cluster_name",
+            gcpGkeContainerMonitoredResource.getClusterName());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "container_name",
+            gcpGkeContainerMonitoredResource.getContainerName());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder,
+            resourceType,
+            "namespace_id",
+            gcpGkeContainerMonitoredResource.getNamespaceId());
+        putResourceAttributeMapToBuilder(
+            attributesBuilder, resourceType, "pod_id", gcpGkeContainerMonitoredResource.getPodId());
+        return;
+    }
+  }
+
+  private static void putResourceAttributeMapToBuilder(
+      Attributes.Builder attributesBuilder,
+      ResourceType resourceType,
+      String attributeName,
+      String attributeValue) {
+    attributesBuilder.putAttributeMap(
+        createResourceLabelKey(resourceType, attributeName),
```

Comment:
We should cache all of these work.
private static final java.util.Map<java.lang.String, com.google.devtools.cloudtrace.v2.AttributeValue> GCE_INSTANCE_LABELS;


attributesBuilder.putAllAttributeMap(GCE_INSTANCE_LABELS);

Reply:
Make sense to me, though making static map makes it difficult to test. 

If we really want to test the resource labels under all 3 cases, we may need to use something like PowerMock to mock `MonitoredResourceUtils.getDefaultResource()`. @sebright any idea on this?

Reply:
If it is better to cache the whole map, couldn't we just pass RESOURCE_LABELS into generateSpan, instead of passing in RESOURCE?

Reply:
Ahh you're right, updated. I also left a TODO to test the parsing.